File: baseble/src/main/java/com/fcbox/ble/baseble/callback/scan/ListFilterScanCallback.java
package com.fcbox.ble.baseble.callback.scan;


import com.fcbox.ble.baseble.model.BluetoothLeDevice;

import java.util.List;

/**
 * 指定设备集合进行过滤，一般用设备名称和 Mac 地址集合
 */
@SuppressWarnings("unused")
public class ListFilterScanCallback extends ScanCallback {
    private List<String> deviceNameList;//指定设备名称集合
    private List<String> deviceMacList;//指定设备 Mac 地址集合

    public ListFilterScanCallback(IScanCallback scanCallback) {
        super(scanCallback);
    }

    public ListFilterScanCallback setDeviceNameList(List<String> deviceNameList) {
        this.deviceNameList = deviceNameList;
        return this;
    }

    public ListFilterScanCallback setDeviceMacList(List<String> deviceMacList) {
        this.deviceMacList = deviceMacList;
        return this;
    }

    @Override
    public BluetoothLeDevice onFilter(BluetoothLeDevice bluetoothLeDevice) {
        BluetoothLeDevice tempDevice = null;
        if (deviceNameList != null && deviceNameList.size() > 0) {
            for (String deviceName : deviceNameList) {
                if (bluetoothLeDevice != null && bluetoothLeDevice.getName() != null && deviceName != null
                        && deviceName.equalsIgnoreCase(bluetoothLeDevice.getName().trim())) {
                    tempDevice = bluetoothLeDevice;
                }
            }
        } else if (deviceMacList != null && deviceMacList.size() > 0) {
            for (String deviceMac : deviceMacList) {
                if (bluetoothLeDevice != null && bluetoothLeDevice.getAddress() != null && deviceMac != null
                        && deviceMac.equalsIgnoreCase(bluetoothLeDevice.getAddress().trim())) {
                    tempDevice = bluetoothLeDevice;
                }
            }
        }
        return tempDevice;
    }
}

File: baseble/src/main/java/com/xzy/ble/baseble/callback/scan/ListFilterScanCallback.kt
package com.xzy.ble.baseble.callback.scan

import com.xzy.ble.baseble.model.BluetoothLeDevice


/**
 * 指定设备集合进行过滤，一般用设备名称和 Mac 地址集合
 */
@Suppress("unused")
class ListFilterScanCallback(scanCallback: IScanCallback) : ScanCallback(scanCallback) {
    private var deviceNameList: List<String>? = null//指定设备名称集合
    private var deviceMacList: List<String>? = null//指定设备 Mac 地址集合

    fun setDeviceNameList(deviceNameList: List<String>): ListFilterScanCallback {
        this.deviceNameList = deviceNameList
        return this
    }

    fun setDeviceMacList(deviceMacList: List<String>): ListFilterScanCallback {
        this.deviceMacList = deviceMacList
        return this
    }

    override fun onFilter(bluetoothLeDevice: BluetoothLeDevice): BluetoothLeDevice? {
        var tempDevice: BluetoothLeDevice? = null
        if (deviceNameList != null && deviceNameList!!.isNotEmpty()) {
            for (deviceName in deviceNameList!!) {
                if (deviceName.equals(bluetoothLeDevice.name.trim { it <= ' ' }, ignoreCase = true)) {
                    tempDevice = bluetoothLeDevice
                }
            }
        } else if (deviceMacList != null && deviceMacList!!.isNotEmpty()) {
            for (deviceMac in deviceMacList!!) {
                if (deviceMac.equals(bluetoothLeDevice.address.trim { it <= ' ' }, ignoreCase = true)) {
                    tempDevice = bluetoothLeDevice
                }
            }
        }
        return tempDevice
    }
}
